docs/mkdocs/docs/technical/contributing.md

```diff
@@ -0,0 +1,195 @@
+
+
+Common Tasks
+============
+
+## Recursive checkout
+```bash
+git clone https://github.com/man-group/ArcticDB.git
+cd ArcticDB
+git submodule init && git submodule update
+```
+
+## CMake presets
+To make it easier to set and share all the environment variables, config and commands, we recommend using the [CMake
+presets](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html) feature.
+
+Recent versions of some popular C++ IDEs support reading/importing these presets:
+* [Visual Studio & Code](https://devblogs.microsoft.com/cppblog/cmake-presets-integration-in-visual-studio-and-visual-studio-code/)
+* [CLion](https://www.jetbrains.com/help/clion/cmake-presets.html)
+
+And it's equally easy to use on the command line ([see example](#compiling-step-by-step)).
+
+We already ship a `CMakePresets.json` in the cpp directory, which is used by our builds.
+You can add a `CMakeUserPresets.json` in the same directory for local overrides.
+Inheritance is supported.
+
+Example `CMakeUserPresets.json`:
+
+<details>
+<summary>Windows Preset to specify a Python version</summary>
+
+```json
+{
+  "version": 3,
+  "configurePresets": [
+    {
+      "name": "alt-vcpkg-debug:py3.10",
+      "inherits": "windows-cl-debug",
+      "cacheVariables": {
+        "Python_ROOT_DIR": "C:\\Program Files\\Python310"
+      },
+      "environment": {
+        "PATH": "C:\\Users\\me\\AppData\\Roaming\\Python\\Python310\\Scripts;C:\\Program Files\\Python310;$penv{PATH}",
+        "PYTHONPATH": "C:\\Program Files\\Python310\\Lib;C:\\Users\\me\\AppData\\Roaming\\Python\\Python310\\site-packages"
+      }
+    }
+  ],
+  "buildPresets": [
+    {
+      "name": "alt-vcpkg-debug:py3.10",
+      "configurePreset": "alt-vcpkg-debug:py3.10",
+      "inheritConfigureEnvironment": true
+    }
+  ]
+}
+```
+</details>
+
+## vcpkg caching
+We use [vcpkg](https://vcpkg.io/) to manage the C++ dependencies.
+
+The CMakePresets we provided use the `cpp/vcpkg` directory as the toolchain, so
+it's not necessary to separately download and install vcpkg.
+(You can, however, maintain a separate vcpkg install and override the CMake presets.)
+
+This setup downloads the dependencies' source packages from the Internet and compiles them locally during CMake
+configuration, so it's critical to cache this process to save time (and to comply with any local IT policy).
+
+Please refer to below documentation:
+* [Binary caching](https://learn.microsoft.com/en-us/vcpkg/users/binarycaching)
+* [Asset caching](https://learn.microsoft.com/en-us/vcpkg/users/assetcaching)
+
+For example, you can use a network share as the caching folder and put the caching configuration environment
+variables into the CMake presets.
+
+Please also note compiling the dependencies uses a lot of disk space.
+Once CMake configuration is done and all the dependencies saved into the binary cache,
+you can remove the `cpp\vcpkg\buildtrees` folder.
+
+## pybind11 configuration
```

Comment:
I don't think we should write so much about this. We should just describe how to flip between Python versions with `Python_ROOT_DIR` and have that be our first-class approach.

Reply:
It is the part that is the most poorly documented by third-party tools and most likely to go wrong.

Before I figure out what exactly is happening, I found probably 30 different ways to do it wrong. Including compiling against Python 3.10 headers but linking to 3.6 library (the most confusing errors).

I doubt a typical develop will do it right the first time without guidance.

Reply:
Or really good cmakepresets with a well documented single thing to tweak?